e2e/nomostest/gitproviders/git-provider.go

```diff
@@ -44,29 +43,8 @@ type GitProvider interface {
 	DeleteObsoleteRepos() error
 }
 
-// GitProviderOpt is an optional parameter for instantiating a new GitProvider
-type GitProviderOpt func(opts *GitProviderOpts)
-
-// GitProviderOpts is the set of optional parameters for instantiating a new GitProvider
-type GitProviderOpts struct {
-	portForwarder *portforwarder.PortForwarder
-}
-
-// WithPortForwarder provides a PortForwarder for the GitProvider to use.
-// Required for LocalProvider in order to establish a PortForwarder to the in-cluster
-// git server.
-func WithPortForwarder(portForwarder *portforwarder.PortForwarder) GitProviderOpt {
-	return func(opts *GitProviderOpts) {
-		opts.portForwarder = portForwarder
-	}
-}
-
 // NewGitProvider creates a GitProvider for the specific provider type.
-func NewGitProvider(t testing.NTB, provider string, opts ...GitProviderOpt) GitProvider {
-	options := GitProviderOpts{}
-	for _, opt := range opts {
-		opt(&options)
-	}
+func NewGitProvider(t testing.NTB, provider string) GitProvider {
```

Comment:
This isn't a constructor. It's a test specific initializer. Should be a method on NT. But I'll fix it later.